modules/@apostrophecms/piece-type/ui/apos/components/AposImageCropper.vue

```diff
@@ -74,27 +98,276 @@ export default {
+     */
+    onFocalPointMouseUp() {
+      const { focalPoint } = this.$refs;
+
+      focalPoint.style.cursor = 'grab';
+      focalPoint.style.transitionDuration = '0.1s';
+
+      document.removeEventListener('mousemove', this.onFocalPointMouseMove);
+      document.removeEventListener('mouseup', this.onFocalPointMouseUp);
+    },
+    /**
+     * Place focal point at the position where the image has been clicked
+     * and update its new coordinates.
+     * Position it in relation of the current target offset
+     * in order to set the `left` and `top` properties accordingly.
+     * Do not place focal point when cropper is changing (move or resize)
+     * to keep a friendly user experience.
+     */
+    onImageClick (event) {
+      if (this.isChangingCropper) {
+        return;
+      }
+
+      const { focalPoint } = this.$refs;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const left = event.clientX - event.currentTarget.offsetLeft - event.currentTarget.offsetParent.offsetLeft - focalPointSize.halfWidth;
+      const top = event.clientY - event.currentTarget.offsetTop - event.currentTarget.offsetParent.offsetTop - focalPointSize.halfHeight;
+
+      focalPoint.style.left = `${left}px`;
+      focalPoint.style.top = `${top}px`;
+
+      this.updateFocalPointCoordinatesDebounced();
+    },
+    /**
+     * Place the focal point back to its position (inside the stencil),
+     * when resizing the screen so that the percentages
+     * relative to the stencil remain the same.
+     */
+    onScreenResize () {
+      this.storeStencilCoordinates();
+      this.placeFocalPointInStencil();
+    },
+    /**
+     * Update stencil and focal point coordinates
+     * after cropper has changed and emit its new coordinates.
+     */
+    handleCropperChange (coordinates) {
+      this.isChangingCropper = false;
+
       if (
-        !this.isUpdatingCoordinates &&
-        this.checkCoordinatesDiff(coordinates, this.docFields.data)
+        !this.isUpdatingCropperCoordinates &&
+        this.checkCropperCoordinatesDiff(coordinates, this.docFields.data)
       ) {
+        this.storeStencilCoordinates();
+        this.updateFocalPointCoordinates();
+
         this.$emit('change', coordinates, false);
       }
 
-      this.isUpdatingCoordinates = false;
+      this.isUpdatingCropperCoordinates = false;
+    },
+    /**
+     * Set cropper coordinates via its API.
+     */
+    setCropperCoordinates (coordinates) {
+      this.$refs.cropper.setCoordinates(coordinates);
+    },
+    /**
+     * Get focal point relative position inside the stencil
+     * and emit it to update `x` and `y` as percentages,
+     * or as `null` if outside it.
+     */
+    updateFocalPointCoordinates () {
+      const { focalPoint } = this.$refs;
+
+      const focalPointSize = this.getFocalPointSize();
+
+      const x = (focalPoint.offsetLeft + focalPointSize.halfWidth - this.stencilCoordinates.left) / this.stencilCoordinates.width;
```

Comment:
Indent if possible

Reply:
IMO, this is much harder to read:

```js
      const x = (
        focalPoint.offsetLeft +
        focalPointSize.halfWidth -
        this.stencilCoordinates.left
      ) / this.stencilCoordinates.width;
      const y = (
        focalPoint.offsetTop +
        focalPointSize.halfHeight -
        this.stencilCoordinates.top
      ) / this.stencilCoordinates.height;
```

I can go with:

```js
      const x =
        (focalPoint.offsetLeft + focalPointSize.halfWidth - this.stencilCoordinates.left) / this.stencilCoordinates.width;
      const y =
        (focalPoint.offsetTop + focalPointSize.halfHeight - this.stencilCoordinates.top) / this.stencilCoordinates.height;
```

but I don't see a lot of differences...

We should have a eslint rule that enforces us to respect the limit of characters

Reply:
Yes we should 👍🏼